RecoParticleFlow/PFProducer/src/PFEGammaAlgo.cc

```diff
@@ -88,11 +88,11 @@ namespace {
                                const size_t test,
                                const float EoPin_cut = 1.0e6) {
     constexpr reco::PFBlockElement::TrackType ConvType = reco::PFBlockElement::T_FROM_GAMMACONV;
+    auto elemkey = &(block->elements()[key]);
     // this is inside out but I just want something that works right now
-    switch (keytype) {
+    switch (elemkey->type()) {
       case reco::PFBlockElement::GSF: {
-        const reco::PFBlockElementGsfTrack* elemasgsf =
-            docast(const reco::PFBlockElementGsfTrack*, &(block->elements()[key]));
+        const reco::PFBlockElementGsfTrack* elemasgsf = docast(const reco::PFBlockElementGsfTrack*, elemkey);
         if (elemasgsf && valtype == PFBlockElement::ECAL) {
           const ClusterElement* elemasclus = reinterpret_cast<const ClusterElement*>(&(block->elements()[test]));
```

Comment:
reinterpret_cast should be avoided. I'd expect static_cast should work here and be a much better choice.

Reply:
Based on a very fast look at the file, it like all uses of reinterpret_cast in this file should be replaced with static_cast

Reply:
Thank you. I will look into it.

Reply:
I changed all reinterpret_cast to static_cast. I am wondering about dynamic_cast: ttps://github.com/cms-sw/cmssw/blob/master/RecoParticleFlow/PFProducer/src/PFEGammaAlgo.cc#L2002 should stay as it is or not.

Reply:
Maybe replace it with the `docast()` function, so that `dynamic_cast` behaviour is retained in debug mode...

Reply:
The one remaining dynamic_cast isn't very helpful since there is no check that the value isn't null. So either docast or static_cast makes sense as a replacement.

Reply:
Thank you @perrotta @Dr15Jones. I adopted docast.